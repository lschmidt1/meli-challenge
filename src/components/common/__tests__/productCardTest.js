import { shallow, mount, render } from "enzyme";
import React from "react";
import { BrowserRouter } from "react-router-dom";
import ProductCard from "../productCard";

describe("<ProductCard />", () => {
  let component;

  beforeEach(() => {
    component = render(
      <BrowserRouter>
        <ProductCard />
      </BrowserRouter>
    );
  });
  describe("should render correctly", () => {
    test("by matching snapshot", () => {
      expect(component).toMatchSnapshot();
    });
  });

  it("doesn't render without props", () => {
    component = mount(
      <BrowserRouter>
        <ProductCard />
      </BrowserRouter>
    );
    expect(component.find("#productCard").exists()).toBe(false);
  });

  it("renders with product props", () => {
    const product = {};
    component = mount(
      <BrowserRouter>
        <ProductCard product={product} />
      </BrowserRouter>
    );
    expect(component.find("#productCard").exists()).toBe(true);
  });
});
