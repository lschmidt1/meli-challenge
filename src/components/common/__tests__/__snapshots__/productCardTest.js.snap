// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductCard /> should render correctly by matching snapshot 1`] = `
Object {
  "length": 0,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
